### DISCUSSION TITLE
Is there a way to call C# method for string?

### DISCUSSION THREAD

**Discussion by sea0920** [sebastienros/jint#1551] [Q&A]
The following code fails at `str.Trim()`. I guess it's because `str` is converted from C# String to JsString and JsString only has trim() method.

>     public class Program
>     {
>         public static void Main()
>         {
>             var engine = new Engine();
>             engine.SetValue("str", " abc ");
>             var res = engine.Evaluate("str.Trim()");
>             Console.WriteLine(res);
>         }
>     }

I want to expose all the C# String methods. Is there an easy way to do that?

**Comment by lahma** [comment]
You can always add functions to the prototype. Sounds odd that you want to do C# scripting with JS engine though.

**Reply by sea0920** [reply]
It was possible in Jint v1 and people in my company want backwards compatibility for existing scripts.
